tests/testthat/test-epi_slide.R

```diff
@@ -677,3 +955,385 @@ test_that("epi_slide gets correct ref_time_value when groups have non-overlappin
+  # provided, but `epi_slide` can't
+  epi_data <- generate_special_date_data(weeks) %>%
+    group_by(geo_value)
+  result2 <- epi_slide_mean(epi_data,
+    col_names = c("a", "b"), na.rm = TRUE,
+    before = 6L, after = 0L
+  )
+  expect_identical(select(ref_result, -time_value), select(result2, -time_value))
+})
+
+test_that("special time_types without time_step fail in epi_slide_mean", {
+  n_obs <- 6
+  k <- 1:n_obs
+
+  day_times_minute <- lubridate::ydm_h("2022-01-01-15") + lubridate::minutes(k) # needs time_step = lubridate::minutes
+  day_times_hour <- lubridate::ydm_h("2022-01-01-15") + lubridate::hours(k) # needs time_step = lubridate::hours
+
+  # Not supported
+  custom_dates <- c(
+    "January 1, 2022", "January 2, 2022", "January 3, 2022",
+    "January 4, 2022", "January 5, 2022", "January 6, 2022"
+  )
+  not_dates <- c("a", "b", "c", "d", "e", "f")
+
+  test_time_type_mean <- function(dates, before = 6L, after = 0L, ...) {
+    epi_data <- epiprocess::as_epi_df(tibble(
+      geo_value = "al",
+      time_value = dates,
+      a = 1:n_obs
+    ))
+
+    expect_error(
+      epi_slide_mean(epi_data,
+        col_names = "a",
+        before = before, after = after
+      ),
+      class = "epiprocess__epi_slide_mean__unmappable_time_type"
+    )
+  }
+
+  test_time_type_mean(custom_dates)
+  test_time_type_mean(not_dates)
+  test_time_type_mean(day_times_minute)
+  test_time_type_mean(day_times_hour)
+})
+
+test_that("helper `full_date_seq` returns expected date values", {
+  n <- 6L # Max date index
+  m <- 1L # Number of missing dates
+  n_obs <- n + 1L - m # Number of obs created
+  k <- c(0L:(n - (m + 1L)), n) # Date indices
+
+  set.seed(0)
+  rand_vals <- rnorm(n_obs)
+
+  generate_special_date_data <- function(date_seq, ...) {
+    epiprocess::as_epi_df(rbind(tibble(
+      geo_value = "al",
+      time_value = date_seq,
+      a = seq_along(date_seq),
+      b = rand_vals
+    ), tibble(
+      geo_value = "ca",
+      time_value = date_seq,
+      a = rev(seq_along(date_seq)),
+      b = rand_vals + 10
+    ), tibble(
+      geo_value = "fl",
+      time_value = date_seq,
+      a = rev(seq_along(date_seq)),
+      b = rand_vals * 2
+    )), ...)
+  }
+
+  # Basic time type
+  days <- as.Date("2022-01-01") + k
+
+  # Require lubridate::period function to be passed as `time_step`
+  day_times_minute <- lubridate::ydm_h("2022-01-01-15") + lubridate::minutes(k) # needs time_step = lubridate::minutes
+  day_times_hour <- lubridate::ydm_h("2022-01-01-15") + lubridate::hours(k) # needs time_step = lubridate::hours
+  weeks <- as.Date("2022-01-01") + 7L * k # needs time_step = lubridate::weeks
+
+  # Don't require a `time_step` fn
+  yearweeks <- tsibble::yearweek(10L + k)
+  yearmonths <- tsibble::yearmonth(10L + k)
+  yearquarters <- tsibble::yearquarter(10L + k)
+  years <- 2000L + k # does NOT need time_step = lubridate::years because dates are numeric, not a special date format
+
+  before <- 2L
```

Comment:
might be worth checking different values of before and after for `full_date_seq`?